src/test/kotlin/com/esosa/f5pi_backend/services/implementations/UserServiceTest.kt

```diff
@@ -0,0 +1,333 @@
+package com.esosa.f5pi_backend.services.implementations
+
+import com.esosa.f5pi_backend.controllers.requests.UpdateUserRequest
+import com.esosa.f5pi_backend.controllers.responses.FieldResponse
+import com.esosa.f5pi_backend.controllers.responses.GameResponse
+import com.esosa.f5pi_backend.controllers.responses.PlayerResponse
+import com.esosa.f5pi_backend.controllers.responses.SeasonResponse
+import com.esosa.f5pi_backend.data.enums.Role
+import com.esosa.f5pi_backend.data.models.Field
+import com.esosa.f5pi_backend.data.models.Season
+import com.esosa.f5pi_backend.data.models.User
+import com.esosa.f5pi_backend.data.repositories.IUserRepository
+import com.esosa.f5pi_backend.services.interfaces.IFieldService
+import com.esosa.f5pi_backend.services.interfaces.IGameService
+import com.esosa.f5pi_backend.services.interfaces.IPlayerService
+import com.esosa.f5pi_backend.services.interfaces.ISeasonService
+
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.InjectMocks
+import org.mockito.Mock
+import org.mockito.Mockito.`when`
+import org.mockito.junit.jupiter.MockitoExtension
+import org.springframework.data.domain.PageImpl
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.util.Optional
+import java.util.UUID
+
+@ExtendWith(MockitoExtension::class)
+class UserServiceTest {
+
+
+    @Mock
+    lateinit var userRepository: IUserRepository
+
+    @Mock
+    lateinit var gameService: IGameService
+
+    @Mock
+    lateinit var fieldService: IFieldService
+
+    @Mock
+    lateinit var seasonService: ISeasonService
+
+    @Mock
+    lateinit var playerService: IPlayerService
+
+    @InjectMocks
+    lateinit var userService: UserService
+
+    @Test
+    fun saveUser() {
+        val user = User(
+            username = "testUser",
+            password = "testPassword",
+            fullName = "testFullName",
+            email = "test@email.com",
+            role = Role.USER,
+        )
```

Comment:
Este fragmento de código veo que se repite bastante. Podés moverlo a una constante a nivel de la clase junto a los otros objetos que venís declarando (y repitiendo)
```suggestion
class UserServiceTest {
    object UserServiceTestData {
           val TEST_USER = User(
            username = "testUser",
            password = "testPassword",
            fullName = "testFullName",
            email = "test@email.com",
            role = Role.USER,
        )
        val TEST_FIELD = Field(
            name = "testField",
            user = user,
            id = UUID.randomUUID(),
        )
        // ...
    }
}
```